internal/provider-config/config.go

```diff
@@ -266,185 +350,213 @@ func (p *LxdProviderConfig) createLxdServerClient(remote LxdProviderRemoteConfig
 
 			// Create new certificate.
 			errCert := instServer.CreateCertificate(req)
-			if errCert != nil {
-				// If request to create a certificate failed, refresh the
-				// server and check again whether the server is trusted.
-				server, _, err = instServer.GetServer()
-				if err != nil {
-					return err
-				}
 
-				if server.Auth != "trusted" {
-					return fmt.Errorf("Unable to authenticate with remote server: %v", errCert)
-				}
+			// Refresh the server and check again whether the server is trusted.
+			apiServer, _, err = instServer.GetServer()
+			if err != nil {
+				return nil, err
 			}
+
+			if apiServer.Auth != "trusted" {
+				return nil, fmt.Errorf("Unable to authenticate with remote server: %v", errCert)
+			}
+		}
+
+		// Ensure LXD version is supported by the provider.
+		serverVersion := apiServer.Environment.ServerVersion
+
+		ok, err := utils.CheckVersion(serverVersion, supportedLXDVersions)
+		if err != nil {
+			return nil, err
+		}
+
+		if !ok {
+			return nil, fmt.Errorf("LXD server with version %q does not meet the required version constraint: %q", serverVersion, supportedLXDVersions)
 		}
+
+		// Cache initialized server.
+		remote.server = instServer
 	}
 
-	return nil
+	p.remotes[remoteName] = remote
+	return remote.server, nil
 }
 
-// fetchServerCertificate will attempt to retrieve a remote LXD server's
-// certificate and save it to the servercerts path.
-func (p *LxdProviderConfig) fetchLxdServerCertificate(remoteName string) error {
-	lxdRemote := p.getLxdConfigRemote(remoteName)
+// setRemote validates the remote and stores it into the provider config with a given name
+// overwriting any existing remote.
+func (p *LxdProviderConfig) setRemote(remoteName string, remote LxdRemote) error {
+	p.mux.Lock()
+	defer p.mux.Unlock()
 
-	certificate, err := lxd_shared.GetRemoteCertificate(lxdRemote.Addr, "terraform-provider-lxd/2.0")
-	if err != nil {
-		return err
+	// Validate remote.
+	if remoteName == "" {
+		return fmt.Errorf("Remote name cannot be empty")
```

Comment:
```suggestion
		return errors.New("Remote name cannot be empty")
```
Should we use what we do in LXD when there is no format specifier?